experimental/personnames/src/provider.rs

```diff
@@ -0,0 +1,115 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+#![warn(unused_imports)]
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+
+use icu_provider::prelude::*;
+use zerovec::VarZeroVec;
+
+/// This is the equivalent of
+/// https://www.unicode.org/reports/tr35/tr35-personNames.html#personnames-element
+///
+/// There are some liberties taken from the DTD definition,
+/// as we enforce uniqueness on some tags where the documentation doesn't specify any of these
+/// constraints.
+///
+/// e.g. : initialPattern has no upper bound, DTD allows for the element to be specified any number
+/// of times, while in this implementation we are restraining it to the 2 documented types
+/// (`initial`, `initialSequence`).
+#[icu_provider::data_struct(PersonNamesFormattingDefinitionV1Marker = "personnames/personnames@1")]
+#[derive(PartialEq, Clone)]
+#[cfg_attr(feature = "datagen",
+derive(serde::Serialize, databake::Bake),
+databake(path = icu_personnames::provider))
+]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct PersonNamesFormattingDefinitionV1<'data> {
+    /// <nameOrderLocales order="surnameFirst">ko vi yue zh</nameOrderLocales>
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub surname_first_locales: VarZeroVec<'data, str>,
```

Comment:
note to self: verifying that we store Locales as `str` in data [elsewhere](https://unicode-org.github.io/icu4x/docs/icu/locid_transform/provider/struct.AliasesV1.html)

Reply:
Should I change the type to something already existing ? I kind of defaulted to str, but if there's an already existing structure for a list of locale, I'm happy to reuse it.